### ISSUE TITLE
Issue with typing.Self in nested function

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#3151]
**Describe the bug**
Pyright will complain about the use of `typing.Self` inside nested functions where the first parameter is not annoated as `Self`.

**To Reproduce**

See the code below, pyright returns the following:

`test.py:10:32 - error: "Self" cannot be used in a function with a `self` or `cls` parameter that has a type annotation other than "Self" (reportGeneralTypeIssues)`
`test.py:10:41 - error: "Self" cannot be used in a function with a `self` or `cls` parameter that has a type annotation other than "Self" (reportGeneralTypeIssues)`

**Expected behavior**
The code should type-check

**Screenshots or Code**
```python
from typing import Callable
from typing_extensions import Self


class Bar:

    @classmethod
    def bar(cls) -> Callable[[int, Self], Self]:
        
        def inner(_: int, bar: Self) -> Self:
            return bar

        return inner

```
![image](https://user-images.githubusercontent.com/8677174/156874681-1bf869f5-dd25-4d07-abf0-1206e2d79971.png)

**VS Code extension or command-line**
pyright 1.1.227

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.228, which I just published. It will also be included in the next release of pylance.
